api/v1alpha1/shared_types.go

```diff
@@ -348,6 +348,8 @@ const (
 )
 
 // KubernetesHorizontalPodAutoscalerSpec defines Kubernetes Horizontal Pod Autoscaler settings of Envoy Proxy Deployment.
+// When HPA is enabled, it is recommended that the value in `KubernetesDeploymentSpec.replicas` be removed, otherwise
```

Comment:
check here @Xunzhuo

Reply:
Yes, I mean if eg should take care the priority in code logics. cc @arkodg @zirain

Reply:
+1, HPA will override replicas

Reply:
Prefer if we kept it simple for now, so our API matches K8s behavior